### ISSUE TITLE
Bug: TS2742: The inferred type of 'config' cannot be named without a reference to ...

### ISSUE THREAD

**Issue by Shadow-Devil** [eslint/eslint#20192]
### Environment

Node version: v24.4.1
pnpm version: 10.13.1
Local ESLint version: 9.37.0
Global ESLint version: -
Operating System: Windows 11

### What parser are you using?

@typescript-eslint/parser

### What did you do?

<details>
<summary>Configuration</summary>

eslint.config.ts
```js
import eslint from "@eslint/js";
import { defineConfig } from "eslint/config";

export default defineConfig(
  eslint.configs.recommended,
);
```
</details>

<details>
<summary>TS Config</summary>

tsconfig.json
```js
{
  "compilerOptions": {
    "module": "NodeNext",
    "target": "esnext",
    "lib": ["esnext"],
    "types": ["node"],
    "moduleResolution": "nodenext",
    "sourceMap": true,
    "declaration": true,
    "declarationMap": true,
    "noUncheckedIndexedAccess": true,
    "strict": true,
    "jsx": "react-jsx",
    "verbatimModuleSyntax": true,
    "isolatedModules": true,
    "noUncheckedSideEffectImports": true,
    "moduleDetection": "force",
    "skipLibCheck": true,
    "allowImportingTsExtensions": true,
    "noEmit": true,
    "allowSyntheticDefaultImports": true,
  },
  "include": [
    "src/**/*.ts",
    "eslint.config.ts"
  ],
}

```
</details>
I updated eslint from version 9.36.0 to 9.37.0.

### What did you expect to happen?

No type errors in my config.

### What actually happened?

```
> tsc

eslint.config.ts:4:1 - error TS2742: The inferred type of 'default' cannot be named without a reference to './node_modules/.pnpm/@eslint+core@0.16.0/node_modules/@eslint/core/dist/cjs/types.d.cts'. This is likely not portable. A type annotation is necessary.

4 export default defineConfig(
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
5   eslint.configs.recommended,
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
6 );
  ~~
```

### Link to Minimal Reproducible Example

https://github.com/Shadow-Devil/hogwarts-productivity-hub-bot

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

My "Minimal Reproducible Example" is not minimal but a link to my repo where it happened.

**Comment by fasttime** [comment]
Thanks for the report @Shadow-Devil! This is a known issue, it's being tracked in eslint/rewrite#283.
